File: java/algorithms/easy/staircase.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-06-13
 * Last modified: 2026-06-13
 * @description Staircase - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/staircase/problem}
 */
//---------- solution ----------
class Result
{
    /**
     * Print a staircase of size n using # symbols and spaces.
     * @param {int} n
     */
    public static void staircase(int n)
    {
        for (int i = 1; i <= n - 1; i++)
        { System.out.println(" ".repeat(n - i) + "#".repeat(i)); }

        System.out.println("#".repeat(n));
    }//end staircase()
}//end class

File: go/algorithms/easy/staircase.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Plus Minus - HackerRank
See: https://www.hackerrank.com/challenges/plus-minus/problem
*/
//--------- package ---------
package easy


//--------- Imports ---------
import (
	"fmt"
	"strings"
)


//--------- functions ---------
func staircase(n int32) {
	for i := int32(1); i <= n-1; i++ {
		fmt.Printf("%s%s\n", strings.Repeat(" ", int(n-i)), strings.Repeat("#", int(i)))
	}//end for

	fmt.Println(strings.Repeat("#", int(n)))
}//end staircase()